### DISCUSSION TITLE
Cop to require comment justifying # rubocop:disable

### DISCUSSION THREAD

**Discussion by bessey** [rubocop/rubocop#12552] [Ideas]
I think its pretty unavoidable that for non-stylistic cops, there will be >0 false positives. In such instances we can disable the cop, and so long as we aren't regularly disabling the cop instead of fixing the issue, this isn't a bad thing. 

I believe its commonly consider best practice to justify the disabling of a cop, but AFAIK there is no programmatic way to enforce this. Justifications might not always be _good_ of course, but by forcing it to be written down we can increase the odds that a peer reviewer identifies the flawed logic.

Has such functionality been discussed / implemented before? I could not find evidence of that, but please 
do share them if you are.

Failing that, I propose introducing a formal syntax for justifying, and a cop to enforce the presence of a justification

```ruby
# RUBOCOP: This batch is too big to run validations, and there are none on this attribute
relation.in_batches.update_all(verified: true) # rubocop:disable Rails/SkipsModelValidations
```

Or a slightly nicer syntax (that relies on a [non-existent Rubocop feature](https://github.com/rubocop/rubocop/pull/9559)) might be

```ruby
# rubocop:disable:next Rails/SkipsModelValidations -- This batch is too big to run validations, 
# and there are none on this attribute
relation.in_batches.update_all(verified: true)
```

For existing codebases, there would need to be a way to note the absence of an existing justification when the cop was enabled, e.g.

```ruby
# RUBOCOP: Disabled before Lint/JustifyDisable was enabled
relation.in_batches.update_all(verified: true) # rubocop:disable Rails/SkipsModelValidations
```

or

```ruby
# rubocop:disable:next Rails/SkipsModelValidations -- Disabled before Lint/JustifyDisable was enabled
relation.in_batches.update_all(verified: true)
```

**Comment by To1ne** [comment]
@bessey We wrote our own cop for that: https://gitlab.com/gitlab-org/gitlab/-/blob/master/rubocop/cop/style/inline_disable_annotation.rb
